source/documentation/airflow.md

```diff
@@ -333,10 +346,75 @@ To deploy your Airflow sandbox, follow the instructions in the [Work with Analyt
 
-Once you have deployed your Airflow sandbox, you should store the script for the DAG you want to test in the `airflow/dags` folder in your home directory on the Analytical Platform. Airflow scans this folder every three minutes and will automatically detect your pipeline.
+Once you have deployed your Airflow sandbox, you should store the script for the DAG you want to test in the `airflow/dags` folder in your home directory on the Analytical Platform. Airflow scans this folder every three minutes and will automatically detect your DAG.
 
-To ensure that your pipeline runs correctly, you should set the `ROLE` variable in your DAG script to be your own IAM role on the Analytical Platform. This is your GitHub username in lowercase prefixed with `alpha_user_`. For example, if your GitHub username was `Octocat-MoJ`, your IAM role would be `alpha_user_octocat-moj`.
+To ensure that your DAG runs correctly, you should set the `ROLE` variable in your DAG script to be your own IAM role on the Analytical Platform. This is your GitHub username in lowercase prefixed with `alpha_user_`. For example, if your GitHub username was `Octocat-MoJ`, your IAM role would be `alpha_user_octocat-moj`.
 
 You should also set the `NAMESPACE` variable in your DAG script to be your own namespace on the Analytical Platform. This is your GitHub username in lowercase prefixed with `user-`. For example, if your GitHub username was `Octocat-MoJ`, your namespace would be `user-octocat-moj`.
 
 You should only use your Airflow sandbox for testing purposes.
+
+## Troubleshooting
+
+If a task fails in Airflow, you can click on it and view the logs. The final few lines before `Event with job id your-task-id-9stuff876 Failed` should indicate the problem. As always, it is a good idea to have print statements in your code to help identify the stage at which it failed.
+Here are some common errors you might encounter:
+
+- **DAG doesn't start on schedule** Before initial use, a DAG needs to be switched on using the Airflow UI (toggle in top left). If you used the homepage toggles, please be very careful not to turn off other critical DAGs. The Airflow UI can misbehave a lot and sometimes the homepage toggles are just wrong, so better to use the toggle within a DAG menu.
+
+- **ERROR - Pod Launching failed: Pod took too long to start** Usually this happens when the image referenced in Airflow does not exist. This could be because you haven't made a Github release, or because Concourse has failed to build it, either because of an error in the image or Concourse itself. You can check the status within Concourse (image -> Deploy -> top right) to see the latest build version, and prompt a build with the + button. Brand new images will need to be 'unpaused' in the Concourse UI. Subsequent images should build automatically after a Github release, provided they use an incremental v0.0.0 format. Occasionally this error may happen randomly, in which case you should use task retries.
```

Comment:
Could '(image -> Deploy -> top right)' be made clearer. There may be something on this in another section that you can link to.

Reply:
Certainly. There is some other stuff on concourse, but the document is pretty bloated and fragmented now. I think a better solution would be to restructure the whole thing really, and have dedicated 'Using Concourse / Airflow / etc.' chapters, which are then linked to from within dedicated tutorial chapters.